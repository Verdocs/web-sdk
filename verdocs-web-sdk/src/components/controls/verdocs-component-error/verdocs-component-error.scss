@import '../../../theme.scss';

verdocs-component-error {
  font-family: $verdocs-primary-font;
  display: flex;
  padding: 15px;
  align-items: center;
  justify-content: center;

  .inner {
    flex: 1;
    height: 300px;
    display: flex;
    font-size: 18px;
    box-sizing: border-box;
    padding: 0 20px;
    background: #ffffff;
    align-items: center;
    justify-content: center;
  }
}
