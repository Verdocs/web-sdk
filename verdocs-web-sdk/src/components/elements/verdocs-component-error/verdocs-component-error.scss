@import '../../../theme.scss';

verdocs-component-error {
  font-family: $verdocs-primary-font;
  display: flex;
  padding: 15px;
  align-items: center;
  justify-content: center;

  .inner {
    flex: 1;
    height: 300px;
    display: flex;
    font-size: 18px;
    background: #ffffff;
    align-items: center;
    justify-content: center;
  }
}
