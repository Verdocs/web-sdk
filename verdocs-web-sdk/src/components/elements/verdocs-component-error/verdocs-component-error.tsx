// NOTE: This component does not have a story because it's not intended for external use.

import {Component, h, Host, Prop} from '@stencil/core';

/**
 * Render a simple error message.
 */
@Component({
  tag: 'verdocs-component-error',
  styleUrl: 'verdocs-component-error.scss',
  shadow: false,
})
export class VerdocsComponentError {
  /**
   * The message to display.
   */
  @Prop() message: string = '';

  render() {
    return (
      <Host>
        <div class="inner">{this.message}</div>
      </Host>
    );
  }
}
