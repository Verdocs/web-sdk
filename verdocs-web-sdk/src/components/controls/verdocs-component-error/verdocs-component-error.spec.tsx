import { newSpecPage } from '@stencil/core/testing';
import { VerdocsComponentError } from './verdocs-component-error';

describe('verdocs-component-error', () => {
  it('renders with default props', async () => {
    const page = await newSpecPage({
      components: [VerdocsComponentError],
      html: `<verdocs-component-error></verdocs-component-error>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders with a message', async () => {
    const page = await newSpecPage({
      components: [VerdocsComponentError],
      html: `<verdocs-component-error message="Error occurred"></verdocs-component-error>`,
    });
    expect(page.root).toMatchSnapshot();
    expect(page.root.querySelector('.inner')?.textContent).toBe('Error occurred');
  });
});
