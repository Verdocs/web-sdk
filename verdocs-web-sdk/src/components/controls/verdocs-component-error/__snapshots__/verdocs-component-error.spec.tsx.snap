// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verdocs-component-error renders with a message 1`] = `
<verdocs-component-error message="Error occurred">
  <div class="inner">
    Error occurred
  </div>
</verdocs-component-error>
`;

exports[`verdocs-component-error renders with default props 1`] = `
<verdocs-component-error>
  <div class="inner"></div>
</verdocs-component-error>
`;
